### PR TITLE
feat(slider): add feature targeting for styles

### PR DESCRIPTION
**Pull request by crisbeto**
Adds support for style feature targeting to the `slider` package.

Relates to #4227.

### CODE REVIEW

File: packages/mdc-slider/_mixins.scss [hunk 1]
```diff
@@ -20,116 +20,457 @@
...
+      &:focus {
+        outline: none;
+      }
+    }
+
+    @include mdc-slider-track_($query);
+    @include mdc-slider-track-marker_($query);
+    @include mdc-slider-thumb_($query);
+    @include mdc-slider-focus-ring_($query);
+    @include mdc-slider-pin_($query);
```
Comment (crisbeto) [inline packages/mdc-slider/_mixins.scss]: Note that these were all under one huge selector. I broke them up into mixins so it's easier to follow.

**Comment by crisbeto** [comment]
Here's the before and after output:

[before.css.txt](https://github.com/material-components/material-components-web/files/3352006/before.css.txt)
[after.css.txt](https://github.com/material-components/material-components-web/files/3352007/after.css.txt)

File: packages/mdc-slider/_mixins.scss [hunk 2]
```diff
@@ -20,116 +20,457 @@
...
+    @include mdc-slider-emphasize-keyframes_;
+  }
+
+  .mdc-slider {
+    @include mdc-slider-color-accessible(secondary, $query);
+
+    &--disabled {
+      $disabled-color: #9a9a9a;
+
+      @include mdc-slider-highlight-color_($disabled-color, $query);
```
Comment (abhiomkar) [inline packages/mdc-slider/_mixins.scss]: Let's name the `$query` parameter for all mixins in this file.

`$query: $query`

File: packages/mdc-slider/_mixins.scss [hunk 3]
```diff
@@ -20,116 +20,457 @@
 // THE SOFTWARE.
 //
 
+@import "@material/feature-targeting/functions";
+@import "@material/feature-targeting/mixins";
```
Comment (abhiomkar) [inline packages/mdc-slider/_mixins.scss]: Update package.json with feature targeting dependency.

File: packages/mdc-slider/_mixins.scss (hunk 1 above)
Comment (abhiomkar) [reply packages/mdc-slider/_mixins.scss]: Thanks!

**Comment by crisbeto** [comment]
Addressed the feedback and rebased @abhiomkar. Can you take another look?

**Pull request merged**